### PR TITLE
[Secure JSON RPC] Refactor the lightweight JSON RPC client to support optional versions. 

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This small PR makes the 'version' field optional in the 'get_account_state()' API call of the secure JSON RPC client. This prevents the client from having to "sync up" to the latest state of the blockchain, and reflects recent changes to the JSON RPC server code (see  here: https://github.com/libra/libra/pull/3460).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

All local tests pass (including the newly added one).

## Related PRs

None.

### CODE REVIEW

File: secure/json-rpc/src/lib.rs [hunk 1]
```diff
@@ -63,18 +63,71 @@ impl From<FromHexError> for Error {
...
+
+    // Sends a get_account_state_with_proof() request to the JSON RPC server for the specified
+    // account, optional version height and optional ledger_version height (for the proof).
+    fn get_account_state_with_proof(
+        &self,
+        account: AccountAddress,
+        version: Option<u64>,
+        ledger_version: Option<u64>,
+    ) -> Result<Response, Error> {
+        let method = "get_account_state_with_proof".to_string();
```
Comment (davidiw) [inline secure/json-rpc/src/lib.rs:119]: use into

File: secure/json-rpc/src/lib.rs [hunk 2]
```diff
@@ -63,18 +63,71 @@ impl From<FromHexError> for Error {
...
+            .timeout_connect(REQUEST_TIMEOUT)
+            .send_json(
+                json!({"jsonrpc": JSON_RPC_VERSION, "method": method, "params": params, "id": 0}),
+            )
+    }
+
+    // Sends a submit() request to the JSON RPC server using the given transaction.
+    fn submit_transaction(&self, signed_transaction: SignedTransaction) -> Result<Response, Error> {
+        let method = "submit".to_string();
+        let params = vec![Value::String(hex::encode(lcs::to_bytes(
```
Comment (davidiw) [inline secure/json-rpc/src/lib.rs:105]: seems like something we could make a function for?
  to_value_string<T: Serialize>(value: T)

Comment (davidiw) [reply secure/json-rpc/src/lib.rs:105]: unless it is done only once...

File: secure/json-rpc/src/lib.rs (hunk 1 above)
Comment (JoshLind) [reply secure/json-rpc/src/lib.rs]: Done (and changed all other to_string() to into(), where appropriate). Thanks!

File: secure/json-rpc/src/lib.rs (hunk 2 above)
Comment (JoshLind) [reply secure/json-rpc/src/lib.rs]: This is only done, once, so I've left it as is! Thanks :)

**Comment by JoshLind** [comment]
Thanks, David! As discussed offline, I've updated this PR to only make the version field optional in the secure JSON RPC client and removed the changes that defined a JsonRpcClient interface. 

For external reader context: the JsonRpcClient interface is no longer needed as we'll test the (up and comign) JsonRpcLibraInterface implementation using end to end tests rather than unit tests, thus we don't need to expose the interface.

**Pull request closed without merging**